relayer-cli/src/conclude.rs

```diff
@@ -72,6 +75,32 @@ pub fn exit_with(out: Output) {
     }
 }
 
+/// Exits the program. Useful when a type produces an error which can no longer be propagated, and
+/// the program must exit instead. Any type that implements the `Unrecoverable` trait can pass
+/// control to this method to bail with an error (which must be a `Display`). The implementation
+/// of `Unrecoverable` is blank.
+///
+/// ## Example of use
+/// - Without support from `Unrecoverable`:
+/// ```ignore
+/// let res = ForeignClient::new(chains.src.clone(), chains.dst.clone());
+/// let client = match res {
+///     Ok(client) => client,
+///     Err(e) => return Output::error(format!("{}", e)).exit(),
+/// };
+/// ```
+/// - With support from `Unrecoverable`:
+/// ```ignore
+/// impl Unrecoverable for ForeignClient {}
+///
+/// let client_a = ForeignClient::new(chains.src.clone(), chains.dst.clone())
+///     .unwrap_or_else(exit_with_unrecoverable_error);
+/// ```
+pub fn exit_with_unrecoverable_error<T: Unrecoverable, E: Display>(err: E) -> T {
+    Output::error(format!("{}", err)).exit();
+    T::conclude()
+}
```

Comment:
I think we could write this as:

```suggestion
pub fn exit_with_unrecoverable_error<T, E: Display>(err: E) -> T {
    Output::error(format!("{}", err)).exit();
    unreachable!()
}
```

Once the [`never (!)`](https://doc.rust-lang.org/std/primitive.never.html) type is stabilized, we can adapt `Output::exit` to return `!` and write this as:

```suggestion
pub fn exit_with_unrecoverable_error<E: Display>(err: E) -> ! {
    Output::error(format!("{}", err)).exit()
}
```